x/dualstaking/types/delegate.go

```diff
@@ -37,6 +40,10 @@ func (delegation *Delegation) Equal(other *Delegation) bool {
 	return true
 }
 
+func (delegation *Delegation) IsFirstMonthPassed(currentTimestamp int64) bool {
+	return delegation.Timestamp+MONTH_DURATION <= currentTimestamp
```

Comment:
save the timestamp as time+month so you wont have to do this calculation every time